core/templates/dev/head/pages/exploration-editor-page/editor-navigation/editor-navigation-dropdown.directive.ts

```diff
@@ -0,0 +1,254 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the showing Editor Navigation in small screen.
+ */
+
+require(
+  'components/common-layout-directives/common-elements/' +
+  'loading-dots.directive.ts');
+
+require('domain/utilities/UrlInterpolationService.ts');
+require('services/ContextService.ts');
+require('services/ExplorationFeaturesService.ts');
+require('services/SiteAnalyticsService.ts');
+require('services/UserService.ts');
+require('services/contextual/WindowDimensionsService.ts');
+require('services/EditabilityService.ts');
+require('pages/exploration-editor-page/services/exploration-rights.service.ts');
+require(
+  'pages/exploration-editor-page/services/exploration-warnings.service.ts');
+require('pages/exploration-editor-page/services/router.service.ts');
+require(
+  'pages/exploration-editor-page/feedback-tab/services/thread-data.service.ts');
+require('pages/exploration-editor-page/services/change-list.service.ts');
+require('pages/exploration-editor-page/services/exploration-save.service.ts');
+
+angular.module('oppia').directive('editorNavigationDropdown', [
```

Comment:
I can understand having duplication in the HTML, but we really can't have so much duplication in the JS. Try and find some way that allows the existing code to be reused. Perhaps instead of making a new directive, you can augment the existing one?

Reply:
I tried to reuse the same js files but the main problem is, I have to combine two js files and I am not able to combine them properly.

Reply:
Yep, that's why I suggested seeing whether you can have a single directive instead of two. Then that directive would only have a single JS file.

Reply:
The problem with that is ,the first js and html is used with nav bar collapse which hides in small screens.So combining them would simply hide the dropdown in small screens.

Reply:
But couldn't the navbar collapse be part of a subsection of the directive? I.e. not all the HTML in the directive has to be under "navbar collapse". You could perhaps have some of the HTML in a "navbar collapse" and some not, similar to how one sometimes has ng-if="condition" and ng-if="!condition" in the same directive to show two different things depending on the current "mode". Would an approach along those lines work?

Reply:
I have solved this problem by placing the options outside the collapsible navbar and deleting one of the directives.

Reply:
I don't think this is solved. A solution would entail most of the functions in this JS file being deleted, or most of the functions in the existing JS file being deleted. Merging this PR in the codebase will result in multiple copies of the same functions.

One approach you could take is to perhaps extend the scope of the directive (and the associated HTML template) to the "lowest parent" that covers all of the editor navigation, the exploration save-and-publish buttons, and the editor navigation dropdown. Then all the parts of this template can use the same JS functions, and you would not be duplicating functionality.

```
                    <div class="collapse navbar-collapse oppia-navbar-collapse ng-cloak">
                      <editor-navbar-breadcrumb></editor-navbar-breadcrumb>

                      <editor-navigation></editor-navigation>
                      <exploration-save-and-publish-buttons></exploration-save-and-publish-buttons>
                    </div>
```

Please note that avoiding unnecessary duplication is a fundamental principle in software engineering -- see e.g. the [DRY rule](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).